src/parser/ParsedQuery.cpp

```diff
@@ -189,8 +189,11 @@ void ParsedQuery::parseAliases() {
         a._isAggregate = true;
         size_t pos = inner.find("as");
```

Comment:
Maybe instead of checking for both "AS" and "as", "DISTINCT" and "distinct" we could have an `ad_utility` function(s) that does case-insensitive checks and replace the above `ad_utility::startsWith()` as well as the `string::find()s` below. I think we can't just lowercase `inner` because that may contain variable names which are case sensitive. According to the [spec](https://www.w3.org/TR/sparql11-query/#sparqlGrammar) SPARQL is suppposed to be case-insensitive by the way, see also #17. Long-term we really should write/generate a proper parser based on the grammar though.

Reply:
As a temporary solution  I changed all the `find`s and `startsWith`s to run on a lowercase version of `inner`. This preserves the spelling of variables but makes all keywords case insensitive.
A proper parser for SPARQL sounds like a very good idea though.